src/coreclr/jit/gentree.cpp

```diff
@@ -24298,8 +24298,148 @@ GenTree* Compiler::gtNewSimdNarrowNode(
 #endif // !TARGET_XARCH && !TARGET_ARM64
 }
 
+GenTree* Compiler::gtNewSimdShuffleNodeVariable(
+    var_types type, GenTree* op1, GenTree* op2, CorInfoType simdBaseJitType, unsigned simdSize, bool isUnsafe)
+{
+    assert(IsBaselineSimdIsaSupportedDebugOnly());
+
+    assert(varTypeIsSIMD(type));
+    assert(getSIMDTypeForSize(simdSize) == type);
+
+    assert(op1 != nullptr);
+    assert(op1->TypeIs(type));
+
+    var_types simdBaseType = JitType2PreciseVarType(simdBaseJitType);
+    assert(op2 != nullptr);
+    assert(op2->TypeIs(type));
+    assert(!op2->IsVectorConst());
+
+    assert(varTypeIsByte(simdBaseType));
+
+    GenTree*             retNode = nullptr;
+    GenTree*             cnsNode = nullptr;
+
+    size_t elementSize  = genTypeSize(simdBaseType);
+    size_t elementCount = simdSize / elementSize;
+
+    assert(elementSize == 1);
+
+#if defined(TARGET_XARCH)
+    // duplicate operand 2 for non-isUnsafe implementation later
+    GenTree* op2DupSafe = nullptr;
+    if (!isUnsafe) op2DupSafe = fgMakeMultiUse(&op2);
+
+    if (simdSize == 32)
+    {
+        assert(compIsaSupportedDebugOnly(InstructionSet_AVX2));
+
+        if (compOpportunisticallyDependsOn(InstructionSet_AVX512VBMI_VL))
+        {
+            // swap the operands to match the encoding requirements
+            retNode = gtNewSimdHWIntrinsicNode(type, op2, op1, NI_AVX512VBMI_VL_PermuteVar32x8, simdBaseJitType, simdSize);
+        }
+        else
+        {
+            // TODO-XARCH-CQ: If we have known min/max or set/unset bits for the indices, then we could avoid an Avx2.Shuffle and/or Avx2.Permute2x128 in some cases
+
+            // get the indices, but only keep the high bit and bits used for indexing, and xor the cross-lane bit on the high 128-bit lane part of indices
+            // we begin computing this early as it seems to take the longest to calculate (it can be done in parallel to other operations ideally)
+            // Vector256<byte> indicesXord = (indices ^ Vector256.Create(Vector128.Create((byte)0), Vector128.Create((byte)0x10))) & Vector256.Create((byte)0x9F);
+            simd_t xorCns = {};
+            for (size_t index = 0; index < simdSize; index++) xorCns.u8[index] = static_cast<uint8_t>(index & 0x10);
```

Comment:
Please add braces and do not nest lines like this

This makes the code not only error prone, but incredibly difficult to read and follow what's going on.

We should feel empowered to use whitespace, newlines, and punctuation (braces, parens, etc) to help with overall legibility and maintainability. If there is a particular case of verbose logic, a helper API could be added to do things like "broadcast" a value to all elements of a `simd_t`